### DISCUSSION TITLE
1.17 Upgrade

### DISCUSSION THREAD

**Discussion by Infuser** [eclipse-theia/theia#10069] [General] [answered]
Hi Theia Team

I am trying to upgrade my Theia installation to 1.17 but I am having issues because JavaScript version got changed to ES2017, I have found a migration guide

https://github.com/eclipse-theia/theia/blob/master/doc/Migration.md but I am not expert on webpack this is the webpack config I have added to my custom extension that overrides the about box

`const path = require('path');
module.exports = {
  target: 'node',
  entry: './src/browser/about-frontend-module.ts',
  mode: 'production',
  devtool: 'source-map',
  externals: {
    vscode: 'commonjs vscode' // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
  },
  module: {
    rules: [
      {
        test: /\.css$/i,
        use: ['css-loader'],
        exclude: /node_modules/
      },
      {
        test: /\.(png|jpg|gif|svg)$/i,
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 8192
            }
          }
        ]
      },
      {
        test: /\.(ts|tsx)$/i,
        use: {
          loader: 'babel-loader',
          options: {
            presets: [['@babel/preset-env', { targets: { chrome: '88' } }]]
          },
        },
      }
    ]
  },
  resolve: {
    extensions: ['.js', '.jsx', '.ts', '.tsx'],
    fallback: {
      "fs": false,
      "tls": false,
      "net": false,
      "path": false,
      "zlib": false,
      "http": false,
      "https": false,
      "stream": false,
      "crypto": false,
      "os": false
    }
  },
  output: {
    filename: 'about-frontend-module.js',
    path: path.resolve(__dirname, 'dist'),
    libraryTarget: 'commonjs2',
    devtoolModuleFilenameTemplate: '../[resource-path]'
  }
};
`

**Comment by Infuser** [comment]
But its failing on these 2 lines at runtime, the ES version issue appears to now be resolved though as the code runs

  // rebind the AboutDialog of theia by our custom one
  rebind(AboutDialog).to(AboutOT2TheiaDialog).inSingletonScope();
  rebind(AboutDialogProps).toConstantValue({ title: 'About' });

It says the AboutDIalog is not bound yet so cannot be rebound

Additonal question in the migration guide it says use webpack to transpile to es5, is the webpack setup needed in the extension project or in the theia ide project, sorry its not very clear to me

**Reply by vince-fugnitto** [reply]
@Infuser you should simply update your target, and if it builds successfully you're fine, else you'll need to make sure you custom code can use `es2017`.

An example is the following:

- https://github.com/eclipse-theia/theia-cpp-extensions/commit/adb81f8b36e0dbfc35223f45bc858fec2b1da236#diff-69540381ae31eb557dbe0862c952999d8efe4e4e3b5bc84cc1916f8e440686d6R14-R23

**Reply by Infuser** [reply]
Thanks Vince worked like a charm

**Comment by vince-fugnitto** [comment, marked as answer]
https://github.com/eclipse-theia/theia/discussions/10069#discussioncomment-1297023 should be the answer if anyone else is having issues :+1:
